src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/BaseTest.cs

```diff
@@ -55,6 +49,20 @@ static BaseTest ()
 			}
 		}
 
+		protected static string RunAdbCommand (string command, bool ignoreErrors = true)
+		{
+			string ext = Environment.OSVersion.Platform != PlatformID.Unix ? ".exe" : "";
+			var home = Environment.GetFolderPath (Environment.SpecialFolder.Personal);
+			string adb = Path.Combine (Environment.GetEnvironmentVariable ("ANDROID_SDK_PATH") ?? $"{home}/android-toolchain/sdk", "platform-tools", "adb" + ext);
```

Comment:
I've made changes over at https://github.com/xamarin/monodroid/pull/558 because they use the same code and I would like to get them in first. I'll back port bits after